andela/ah-infinity-stones
# #161966908 Document API with swagger

#### What does this PR do?
- Document API with Swagger .
#### Description of Task to be completed?
- Install django-rest-swagger 2.1.0
- change views to extend generic views
- Add docs url
- Test available api endpoints with documentation
#### How should this be manually tested?
1. `git clone -b ft-document-api-swagger-161966908 https://github.com/andela/ah-infinity-stones` and `$ cd ah-infinity-stones`
2. Create and activate virtual env
- `virtualenv  -p python3 venv`
- `$ source venv/bin/activate`
3. `$ pip3 install -r requirements.txt` to install the dependencies
4. Setup environment variables. Copy all from below and paste them into the .env file. Update the values to your choice.
```export DATABASE_NAME='ah_dev'
export DATABASE_USER='user'
export DATABASE_PASSWORD=pass'
export HOST='localhost'
export PORT='5432'
export DATABASE_TEST='ah_test'
export EMAIL_HOST_USER='username'
export EMAIL_HOST_PASSWORD='pass'
export SECRET_KEY='key-is-secret'
```
Activate the env variables by `$source .env`.
5. Run migrations: `$ python3 manage.py migrate`
6. Run app - `$ python3 manage.py runserver`
7. Access `http://127.0.0.1:8000` through your browser and test by:
-  `POST /api/users/` - Create account with AH
Check your email and activate the account. Login with the endpoint below:
- `POST /api/users/login` - Login to AH

#### What are the relevant pivotal tracker stories?
[#161966908](https://www.pivotaltracker.com/story/show/161966908)
#### Screenshots
<img width="1192" alt="screenshot 2018-12-20 at 13 57 06" src="https://user-images.githubusercontent.com/12385482/50280993-2e7f2700-045f-11e9-977c-8dca94206488.png">
